621.py

```diff
@@ -0,0 +1,20 @@
+class Solution:
+    def leastInterval(self, tasks: List[str], n: int) -> int:
+        task_freqs = Counter(tasks)
+        sorted_freqs = sorted(task_freqs.values(), reverse=True)
+
+        max_freq, max_count = sorted_freqs[0], 1
+
+        for freq in sorted_freqs[1:]:
+            if freq == max_freq:
+                max_count += 1
+            else:
+                break
+
+        part_count = max_freq - 1
```

Comment:
The names of `part_count` and `part_length` don't really make sense to me. They need to be more descriptive.